lib/rubycritic/analysers/smells/flay.rb

```diff
@@ -1,19 +1,20 @@
 # frozen_string_literal: true
 require 'rubycritic/analysers/helpers/flay'
 require 'rubycritic/core/smell'
-require 'rubycritic/colorize'
 
 module RubyCritic
   module Analyser
     class FlaySmells
-      include Colorize
-      def initialize(analysed_modules)
+      def initialize(analysed_modules, logger=nil)
         @analysed_modules = paths_to_analysed_modules(analysed_modules)
+        @logger = logger
         @flay = Flay.new(@analysed_modules.keys)
       end
 
       def run
-        @flay.hashes.each do |structural_hash, nodes|
+        hashes = @flay.hashes
+        skip_logging = @logger.nil?
```

Comment:
That and everything associated to it below can go away with a NullLogger (see my other comment).